ui/pages/groups/index.js

```diff
@@ -22,7 +22,7 @@ function AddGroupsDialog({ setOpen }) {
     try {
       const res = await fetch('/api/groups', {
         method: 'POST',
-        body: JSON.stringify({ name }),
+        body: JSON.stringify({ name: name.trim() }),
```

Comment:
nit:

Rather than trimming the name here and in the error you could do one `setName(name.trim())` before the try/catch block.